src/particles/ImpactXParticleContainer.cpp

```diff
@@ -141,25 +141,52 @@ namespace impactx
                 gid = *it;
             }
         }
-        auto& particle_tile = DefineAndReturnParticleTile(lid, gid, tid);
 
-        auto old_np = particle_tile.numParticles();
-        auto new_np = old_np + np;
-        particle_tile.resize(new_np);
-
-        // Update NextID to include particles created in this function
-        int pid;
-#ifdef AMREX_USE_OMP
-#pragma omp critical (add_beam_nextid)
+    int nthreads = 1;
+#if defined(AMREX_USE_OMP)
+    nthreads = omp_get_max_threads();
 #endif
-        {
-            pid = ParticleType::NextID();
-            ParticleType::NextID(pid+np);
-        }
+
+    const auto& ba = ParticleBoxArray(lid);
+    auto n_logical =  numTilesInBox(ba[gid], true, tile_size);
+
+    if (n_logical < nthreads ) {
+        amrex::Print() << "Too few tiles for the number of OpenMP threads. Parallelization will be poor. \n";
```

Comment:
To avoid stdout flooding, we can use `#include <ablastr/warn_manager/WarnManager.H>`, e.g.,
https://github.com/BLAST-ImpactX/impactx/blob/61b13e453ea5e953b8a5f985539d3e6833cf088f/src/diagnostics/CovarianceMatrixMath.H#L65-L72